chapters/chap03-sec06.tex

```diff
@@ -14,13 +14,19 @@ \section{$\zksnark$ instantiation}\label{instantiation:zksnark}
 %%
 %% The use of the $\constno$-th root of unity supports FFT transforms for fast polynomial manipulation. The requirement that $\constno$ is a power-of-2 dividing $\rBN$ allows FFT transforms to be used for fast evaluation of Langrange polynomials at $\tau$, in terms of a series of encoded powers $\smallset{ \groupenci{\tau}{1} }_{i = 0}^{\constno - 1}$ (as described in Section 3 of "A multi-party protocol for constructing the public parameters of the Pinocchio zk-SNARK" - https://eprint.iacr.org/2017/602.pdf)
 
+Furthermore, we denote by $\inputs \in \FF^{\inpno+1}$ the tuple of variables (i.e.~``circuit wires'') in the algebraic representation of the relation $\REL$, such that:
+\begin{itemize}
+    \item $\inputs_0 = 1_{\FF}$ (the multiplicative identity in $\FF$),
+    \item $(\inputs_1, \ldots, \inputs_{\inpnoprim})$ represent variables in the statement,
+    \item $(\inputs_{\inpnoprim+1}, \ldots, \inputs_{\inpno})$ represent variables in the witness.
```

Comment:
```suggestion
    \item $(\inputs_{\inpnoprim+1}, \ldots, \inputs_{\inpno})$ represent variables in the witness (so-called ``auxiliary input'').
```
We say "auxiliary input" for "witness" which is quite unfortunate, as it is standard only in Ben-Sasson et al's papers. "Witness" is much more often used term, and "auxiliary input" means sth different in crypto-community. I would stick to "witness" in the further writing.

Reply:
AFAICT, "auxiliary input/private input/witness" are used interchangeably to denote the same thing in the ZKP litterature (the ZK standard reference document seems to be doing exactly that for instance: https://github.com/zkpstandard/docs/blob/master/reference.pdf). The notion of auxiliary input isn't too far fetched in the context of circuits in the zk-SNARK setting because the auxiliary inputs (i.e. private input wires) can be seen as the source of non-determinism in the circuit.

Reply:
That being said, I'm happy to stick to "instance/witness" instead of "primary input/auxiliary input" as it's equally clear

Reply:
@mpzajac not sure I understand: do you suggest to add the (`so-called auxiliary input` bit in https://github.com/clearmatics/zeth-specifications/pull/12#discussion_r564417672)? Isn't that in contradiction with your next sentence in the comment:
> We say "auxiliary input" for "witness" which is quite unfortunate, as it is standard only in Ben-Sasson et al's papers. "Witness" is much more often used term, and "auxiliary input" means sth different in crypto-community. I would stick to "witness" in the further writing.

Happy to add the change, but I don't think I understand what you meant here.

Reply:
In the linked document they write "WitnessVar represents an indexed variable of the witness, or the private/auxiliary input, serialized with positive indices. " what suggests (IMHO) that auxiliary input differ (to some extent) from the witness.

"The notion of auxiliary input isn't too far fetched in the context of circuits in the zk-SNARK setting because the auxiliary inputs (i.e. private input wires) can be seen as the source of non-determinism in the circuit."
Yes, OTOH, in the document you linked, it is also mentioned "auxiliary information" that comes e.g. with the relation generation, which also should be considered algorithm's aux. 
It is even worse when one considers ZKIP-based proof systems where it is important to distinguish prover's randomness (which is a part of its auxiliary input) and its witness. 
Furthermore, consider a case with a zk adversary, who is given some random bits and produces (x, w) pair it query the oracle on. In that case it would be a bit odd (IMO) to call "w" the adversary's aux.

Reply:
Re contradiction: in the document we denote the witness by "auxiliary input", thus I thought it would be good to use the same name in the changed part -- I don't think changing all occurences of "auxiliary input" to "witness" makes sense at this point, I believe we should use "witness" in the future.

Reply:
I see, thanks. I'll add your suggestion then, that sounds good.